### DISCUSSION TITLE
使用白名单让任意页面实现【自动进入阅读模式】

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1672] [使用技巧]
## 技巧背景

当开启 **自动进入阅读模式**（下图） 后，**[当某个站存在于适配列表](http://ksria.com/simpread/docs/#/%E7%AB%99%E7%82%B9%E9%80%82%E9%85%8D%E6%BA%90)** 的话，则会自动进入阅读模式。

![image](https://user-images.githubusercontent.com/81074/103851611-0622ef00-505f-11eb-9664-dd2eb033b617.png)

反之，即便当这个站不存在于适配列表的话，则无法使用 **自动进入阅读模式**。

## 问题描述

> 如何将 **不存在适配列表但可以智能识别出正文的站** 自动进入阅读模式？

这类站如：[百家号](https://baijiahao.baidu.com/s?id=1648183731523753801&wfr=spider&for=pc) 或者 [某个博客](https://happyxiao.com/tim-urban/)
这类站的特点：进入后，简悦会在右上角下显示红色 icon ![image](https://user-images.githubusercontent.com/81074/103852540-e68cc600-5060-11eb-8a84-21b9b402f57f.png) 同时使用快捷键可直接进入阅读模式。

## 解决方案

> 将这些站 **[以正则表达式的方式加入到白名单](http://ksria.com/simpread/docs/#/%E7%AB%99%E7%82%B9%E7%BC%96%E8%BE%91%E5%99%A8?id=%e6%ad%a3%e5%88%99%e8%a1%a8%e8%be%be%e5%bc%8f)** 中，如下：

**百度百家号**
`[[/https\:\/\/baijiahao\.baidu\.com\/s\?id=\S+/]]`

**全现在**
`[[/https:\/\/www\.allnow\.com\/post\/\S+/]]`

## 加入位置

1. 选项页 → 高级设定 → 白名单

2. 在需要的页面 → 右键菜单 → 加入到白名单，如下图所示：

    ![image](https://user-images.githubusercontent.com/81074/103864765-04b1f080-5078-11eb-956c-d5bf52f920dd.png)

    > 如右键菜单没有 **将当前页面加入到白名单** 的选项，需下图方式开启：

    ![image](https://user-images.githubusercontent.com/81074/103864928-46429b80-5078-11eb-9674-081c5be9ab99.png)

## 如何编写域名的匹配

详细说明请看  [站点编辑器](http://ksria.com/simpread/docs/#/站点编辑器?id=对应字段) 

### 精准匹配

```
https://xxx.com/post/39411
https://xxx.com/post/1243
https://xxx.com/post/39411
```

### 模糊匹配

> 支持 [minimatch](https://github.com/isaacs/minimatch)

```
http://*.cnbeta.com/articles/*/*.htm
https://www.ithome.com/html/**/*/*.htm
https://github.com/*/*/issues/*
```

### 正则表达式匹配

> 支持 正则表达式，使用前后缀 `[[/` `/]]`

```
[[/https:\\/\\/movie.douban.com\\/subject\\/\\d+\\/?/]]
```

## 注意

当开启 **自动进入阅读模式** 后，白名单会被隐藏（但不影响它的功能），需将此选项关闭。

**Comment by RavenGladiator** [comment]
请教一下csdn和知乎的正则表达式该如何书写呢？研究了一会没捣鼓出来。。

**Reply by Kenshin** [reply]
你是仅需要 CSDN 与知乎自动进入吗？

如果是的话：

- `csdn.net`
- `zhihu.com`

**Comment by cabbagealy** [comment]
https://www.biduoxs.com/

**Reply by Kenshin** [reply]
让它自动进入阅读模式？试试：

`https://www.biduoxs.com/biquge/*/*.html`

**Comment by RavenGladiator** [comment]
感谢！很受启发，问题得到解决！

**Reply by Kenshin** [reply]
哈哈，上次的提问还是 [21年11月份](https://github.com/Kenshin/simpread/discussions/1672#discussioncomment-1715511)，才看到回复... 

新年快乐

**Comment by RavenGladiator** [comment]
哈哈哈没关系～
新年快乐！简悦是我目前所见到的文本搜索结果阅读的最最优解，非常感谢简悦的出现，祝简悦能在新的一年里蒸蒸日上，越来越好！

发自我的iPhone

**Reply by Kenshin** [reply]
哈哈，很高兴能喜欢简悦。

**Comment by RavenGladiator** [comment]
哈哈哈😄 必须五星好评～

发自我的iPhone

**Comment by angel126** [comment]
file:///E:/ 本地这样的HTML文件打开自动进入阅读模式，应该怎么写呢？ 我试了file:///E:/*，file://* 都没有效果？
![image](https://user-images.githubusercontent.com/8450915/150054839-11978269-6e65-479b-a042-fd207282982e.png)
![image](https://user-images.githubusercontent.com/8450915/150054970-e98d0d00-2c17-4baa-b48b-cd68d931ce87.png)

**Reply by Kenshin** [reply]
嗨，本地文件暂时不支持 **白名单** 模式。

**Reply by woodwolfzhu** [reply]
localhost 支持吗

**Reply by Kenshin** [reply]
@woodwolfzhu 

不支持，但支持 `IP` 方式，最好使用 https://github.com/Kenshin/simpread/discussions/4625

**Comment by fangjin5748** [comment]
您好，请问一下我想让https://www.sixthtone.com/news/1009454/can-chinas-skyscraper-capital-come-down-to-earth 自动进入阅读模式，但是 https://www.sixthtone.com 不自动进入应该怎么写呢，
试了 `https://www.sixthtone.com/news/*/*.html` 不成功呀

**Reply by Kenshin** [reply]
嗨，正确的写法是：

`https://www.sixthtone.com/news/*/*`

**Reply by fangjin5748** [reply]
成功了，太感谢了
居然一会就等到回复了，建议在主页写个范例呀，我没专门学过正则表达式，都是看的网上的教程，想半天也想不明白

**Reply by Kenshin** [reply]
搞定就好。

至于你说的问题，可以看下 [站点编辑器](http://ksria.com/simpread/docs/#/站点编辑器?id=对应字段) 其实里面有写的。

不过，我已经把这些内容放到了文档里面。

**Comment by godefy** [comment]
请问加进白名单以后应该如何取消呢? 找了很久也没找到。

**Reply by godefy** [reply]
没找到从哪里可以再次管理已添加的那些白名单。

**Reply by Kenshin** [reply]
嗨，文档说明 [在这里](http://ksria.com/simpread/docs/#/URL%E7%BC%96%E8%BE%91%E5%99%A8?id=%e7%99%bd%e5%90%8d%e5%8d%95)

选项页 → 高级设定 → 全局

![image](https://user-images.githubusercontent.com/81074/154792032-d57ce8ed-3795-48b1-a857-9af4e00b3b92.png)

**Comment by gradypark86** [comment]
作者你好，我想问一下自动聚焦模式下的两个设置项：
- 是否启用手动聚焦模式？
- 如果当前页面为适配站点，是否自动进入聚焦模式？

有什么区别呢？第二项比较好理解，就是适配站点自动进入聚焦模式。但是第一项看起来又和第二项重复了，难道是控制适配站点外的站点自动进入聚焦模式的吗？又好像不是这样的，那么具体是控制什么的呢？

<img width="644" alt="image" src="https://user-images.githubusercontent.com/39909237/191641201-77555c55-dcaa-4b77-9ef2-82680c6a2abc.png">

**Reply by Kenshin** [reply]
嗨，第一项的意思是：使用聚焦模式的快捷键后，会出现框选需要进入聚焦模式的功能。你可以理解为：并不是根据获取正文的方式进入，一般用于：非适配的页面。

**Comment by Chill-Try** [comment]
手动框选后，右上角会提示加入本地适配，能否多设个选项，让用户选择精确匹配还是模糊匹配呢？当然模糊匹配有时是一个 * 通配符，有时是两个，所以可以快捷弹出站点编辑器。最后，本地站点编辑器有一点比较坑，就是相同的站点不会覆盖。例如，我第一次设置好高亮区域后，保存了 https://www.abc.com/*/* 的高亮区域，之后再此设置 https://www.abc.com/*/* 的高亮区域，保存后站点管理器的自定义适配源会有两条 https://www.abc.com/*/* 的配置

**Reply by Kenshin** [reply]
okay 可以考虑。

> 就是相同的站点不会覆盖。

确实需要优化下。
